frontend/src/api/api.test.ts

```diff
@@ -1,102 +1,106 @@
-import { UploadedFile, ACCEPTED_FILE_TYPES } from '../custom/fileUploadUtils'
-import { ChatMessage, ConversationRequest } from '../api/models'
-import { conversationApi } from './api'
+import { ConversationRequest } from "../api/models";
+import { ACCEPTED_FILE_TYPES, UploadedFile } from "../custom/fileUploadUtils";
 
-describe('Test the conversationApi function', () => {
+import { conversationApi } from "./api";
+
+describe("Test the conversationApi function", () => {
   const defaultUploadedFile: UploadedFile = {
-    name: 'default name',
-    contents: 'default contents',
+    name: "default name",
+    contents: "default contents",
     size: 100,
-    extension: 'default extension'
-  }
+    extension: "default extension",
+  };
 
-  const defaultAbortSignal: AbortSignal = new AbortController().signal
+  const defaultAbortSignal: AbortSignal = new AbortController().signal;
 
   const defaultFetchRequest = {
-    body: '',
-    headers: { 'Content-Type': 'application/json', 'conversation-id': '' },
-    method: 'POST',
-    signal: defaultAbortSignal
-  }
+    body: "",
+    headers: { "Content-Type": "application/json", "conversation-id": "" },
+    method: "POST",
+    signal: defaultAbortSignal,
+  };
 
   const createConversationRequestWithUploadedFile = (file?: UploadedFile): ConversationRequest => {
     return {
       messages: [
         {
-          id: 'default id',
-          role: 'default role',
-          content: 'default content',
-          date: 'default date',
-          uploaded_file: file ?? undefined
-        }
-      ]
-    }
-  }
+          id: "default id",
+          role: "default role",
+          content: "default content",
+          date: "default date",
+          uploaded_file: file ?? undefined,
+        },
+      ],
+    };
+  };
 
   beforeEach(() => {
-    global.fetch = jest.fn()
-  })
+    global.fetch = jest.fn();
+    global.structuredClone = jest.fn().mockImplementation((value) => {
+      return JSON.parse(JSON.stringify(value));
+    });
```

Comment:
[dust] could you help me understand why this new line is needed? was `global.structuredClone` not available in the CI environment so needed this mock?